src/sagemaker_sklearn_extension/contrib/taei/README.md

```diff
@@ -0,0 +1,137 @@
+# Tabular AutoEncoder Interpolator
+
+## Overview
+<img src="images/overview.png" alt="overview" height="200">
+
+This library contains implementations of the latent space minority oversampling techniques proposed in [1] for 
+multi-modal data. These oversamplers work by
+
+1. Mapping the multi-modal samples to a dense continuous latent space using an autoencoder
+2. Applying oversampling by interpolation in the latent space
+3. Mapping the synthetic samples back to the original feature space
+
+This framework was shown to be effective in generating high-quality multi-modal synthetic data which then resulted in 
+better prediction quality for downstream tasks.
+
+#### LatentSpaceOversampler
+The interpolator is implemented by `LatentSpaceOversampler` which takes two inputs:
+- `model` - The autoencoder used to map the samples to the latent space and back. Currently, two 
+autoencoders are provided with the package: `AE` which is a vanilla autoencoder and `VAE` which is a variational 
+autoencoder.
+- `base_oversampler` (function) - The oversampling function applied in the latent space. No oversamplers are provided 
+with this package. We have experimented with `SMOTE` from 
+[imbalanced-learn](https://github.com/scikit-learn-contrib/imbalanced-learn) and `polynom_fit_SMOTE` from 
+[smote_variants](https://github.com/analyticalmindsltd/smote_variants). Note that the base oversampler controls the 
+number of minority samples generated
+
+## Installation
+It is recommended to install from PyPI
+```
+pip install sagemaker-scikit-learn-extension[taei]
+
+# For Zsh users: 
+pip install sagemaker-scikit-learn-extension\[taei]\
+```
+
+## Examples
+[imbalanced-learn](https://github.com/scikit-learn-contrib/imbalanced-learn) is required to run the examples below as 
+it provides the dataset and the base oversampler. [smote_variants](https://github.com/analyticalmindsltd/smote_variants)
+is used for polynom_fit_SMOTE which yield superior prediction quality in our experiments. Install these packages by
+```
+pip install imbalanced-learn==0.7 smote-variants
```

Comment:
I believe there are additional packages that can provide the `base_oversampler`. I guess we can provide a dummy oversampler but it would yield subpar results so I prefer not to do it

Reply:
I prefer not to add these as dependencies. The user can use `base_oversampler` as he wishes

Reply:
On a second thought. I have decided to provide a base oversampler with the package. See `StarOversampler`. Thanks for the idea